src/transformers/models/deformable_detr/modeling_deformable_detr.py

```diff
@@ -1435,21 +1437,21 @@ def __init__(self, config: DeformableDetrConfig):
         self.level_embed = nn.Parameter(torch.Tensor(config.num_feature_levels, config.d_model))
 
         if config.two_stage:
-            self.enc_output = nn.Linear(config.d_model, config.d_model)
+            self.enc_output_proj = nn.Linear(config.d_model, config.d_model)
             self.enc_output_norm = nn.LayerNorm(config.d_model)
             self.pos_trans = nn.Linear(config.d_model * 2, config.d_model * 2)
             self.pos_trans_norm = nn.LayerNorm(config.d_model * 2)
         else:
             self.reference_points = nn.Linear(config.d_model, 2)
 
-        # Initialize weights and apply final processing
-        if not config.two_stage:
+        self.post_init()
+
+    def _init_weights(self, module):
+        if not self.config.two_stage:
             nn.init.xavier_uniform_(self.reference_points.weight.data, gain=1.0)
             nn.init.constant_(self.reference_points.bias.data, 0.0)
         nn.init.normal_(self.level_embed)
```

Comment:
Tried to apply [this suggestion](https://github.com/huggingface/transformers/pull/17281#discussion_r874900210), but it breaks the test `DeformableDetrModelTest.test_initialization`.
Pinging @nielsrogge for a better fix